pkgs/development/libraries/togl/default.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, fetchurl
+, libGL
+, tcl
+, tk
+, xorg
+}:
+
+tcl.mkTclDerivation rec {
+  pname = "togl";
+  version = "2.0";
+
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/togl/files/Togl/${version}/Togl${version}-src.tar.gz";
+    sha256 = "sha256-t9SpC7rTrKYY1QXumef9j7BMgp9jIx3aI2D1V7o/dhA=";
+  };
+
+  postPatch = ''
+    sed -i "s|\".*/generic/tclInt.h\"|\"${tcl}/include/tclInt.h\"|" configure
+    sed -i "s|\".*/generic/tkInt.h\"|\"${tk.dev}/include/tkInt.h\"|" configure
+  '';
+
+  configureFlags = [
+    "--with-tcl=${tcl}/lib"
+    "--with-tk=${tk}/lib"
+  ];
+
+  buildInputs = [
+    tcl
+    tk.dev
+    libGL
+    xorg.libX11
+    xorg.libXmu
```

I think `callPackage` knows how to find these X libraries:

```suggestion
    libX11
    libXmu
```